Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -269,13 +315,22 @@ private void onAddCardSubmit(DropInEvent event) {
         showCardDetailsFragment(cardNumber);
     }
 
-    private boolean shouldAddFragment(String tag) {
+    private boolean fragmentAlreadyExists(String tag) {
```

Comment:
This logic is a little confusing to me - it looks like we use this method to determine if we should show a fragment or not, but this method looks like it's checking if a fragment with this tag exists, not if it's currently visible. Is that right, or does findFragmentByTag only return a fragment if it's currently visible?

Reply:
Yeah visibility isn't a concern here, we just don't want to re-create a fragment if one already exists and is managed by the fragment manager. Guards us against those edge cases like a user double-tapping a button; we want to make sure we don't mistakenly put two Card Details fragments on the screen at once.

Reply:
Went ahead and reverted this to `shouldAddFragment` that does read better.